src/app/WriteHandler.cpp

```diff
@@ -188,6 +180,114 @@ CHIP_ERROR WriteHandler::ProcessAttributeDataIBs(TLV::TLVReader & aAttributeData
+        // cluster info will be invalid / wildcard, it is safe ignore CHIP_END_OF_TLV directly.
+
+        err = attributePath.GetNode(&(clusterInfo.mNodeId));
+        if (CHIP_END_OF_TLV == err)
+        {
+            err = CHIP_NO_ERROR;
+        }
+
+        err = attributePath.GetCluster(&(clusterInfo.mClusterId));
+        SuccessOrExit(err);
+
+        err = attributePath.GetAttribute(&(clusterInfo.mAttributeId));
+        SuccessOrExit(err);
+
+        groupId = mpExchangeCtx->GetSessionHandle()->AsGroupSession()->GetGroupId();
+        fabric  = GetAccessingFabricIndex();
+
+        err = attributePath.GetListIndex(&(clusterInfo.mListIndex));
+        if (CHIP_END_OF_TLV == err)
+        {
+            err = CHIP_NO_ERROR;
+        }
+
+        err = element.GetData(&dataReader);
+        SuccessOrExit(err);
+
+        ChipLogDetail(DataManagement,
+                      "Received group attribute write for Group=%" PRIu16 " Cluster=" ChipLogFormatMEI
+                      " attribute=" ChipLogFormatMEI,
+                      groupId, ChipLogValueMEI(clusterInfo.mClusterId), ChipLogValueMEI(clusterInfo.mAttributeId));
+
+        iterator = groupDataProvider->IterateEndpoints(fabric);
+        VerifyOrExit(iterator != nullptr, err = CHIP_ERROR_NO_MEMORY);
+
+        while (iterator->Next(mapping))
+        {
+            if (groupId != mapping.group_id)
+            {
+                continue;
+            }
+
+            clusterInfo.mEndpointId = mapping.endpoint_id;
+
+            if (!clusterInfo.IsValidAttributePath() || clusterInfo.HasAttributeWildcard())
+            {
+                ChipLogDetail(DataManagement,
+                              "Invalid goup attribute wirte for endpoint=%" PRIu16 " Cluster=" ChipLogFormatMEI
+                              " attribute=" ChipLogFormatMEI,
+                              mapping.endpoint_id, ChipLogValueMEI(clusterInfo.mClusterId),
+                              ChipLogValueMEI(clusterInfo.mAttributeId));
+
+                continue;
+            }
+
+            ChipLogDetail(DataManagement,
+                          "Processing goup attribute wirte for endpoint=%" PRIu16 " Cluster=" ChipLogFormatMEI
+                          " attribute=" ChipLogFormatMEI,
+                          mapping.endpoint_id, ChipLogValueMEI(clusterInfo.mClusterId), ChipLogValueMEI(clusterInfo.mAttributeId));
+
+            chip::TLV::TLVReader tmpDataReader(dataReader);
+
+            const ConcreteAttributePath concretePath =
+                ConcreteAttributePath(clusterInfo.mEndpointId, clusterInfo.mClusterId, clusterInfo.mAttributeId);
+
+            MatterPreAttributeWriteCallback(concretePath);
+            err = WriteSingleClusterData(subjectDescriptor, clusterInfo, tmpDataReader, this);
+            MatterPostAttributeWriteCallback(concretePath);
+        }
+
+        iterator->Release();
+
+        if (CHIP_END_OF_TLV == err)
```

Comment:
This block does not make sense to me.  Shouldn't it be outside the `aAttributeDataIBsReader.Next()` loop? Otherwise we will _always_ return error from this function.

Reply:
done